### PR TITLE
Update Wrapper to Support Pegasus 5.0.0

### LINKED ISSUE #48: Upgrade to 5.0.0dev API for making workflows
> we did do 5.0beta1 release that folks should consider moving it. it has a new workflows api . Migration Guide: https://pegasus.isi.edu/docs/5.0.0dev/migration.html . Release notes: https://pegasus.isi.edu/2020/07/27/pegasus-5-0-beta-released/

### PR DESCRIPTION
**Pull request by LichMaster98**
Closes #48

### CODE REVIEW

**Comment by LichMaster98** [comment]
I believe there are still a few pending changes here including:
-  updating the resource requests to follow the outline of [this page from pegasus' documentation](https://pegasus.isi.edu/docs/5.0.0/user-guide/execution-environments.html?highlight=glite#setting-job-requirements) as we currently add all these parameters as extra parameters when we should adapt to using the standard profiles for all the non-extra types. 
- Troubleshooting the current unit test failures which are currently because we're using an XML parser to verify we've written the DAX correctly when the DAX representation has been changed to YAML
- Verifying a proper install of Pegasus 5.0.0 on `saga03` for current testing -- When trying to run a test workflow I'm currently unable to submit as it appears the appropriate 'glite' configuration may not be complete. See [here for information about what may be required.](https://pegasus.isi.edu/docs/5.0.0/user-guide/execution-environments.html?highlight=glite#example-installation-on-a-rhel-7-login-node)

File: README.md [hunk 1]
```diff
@@ -57,11 +57,6 @@ Use run_on_single_node parameter when you initialize a workflow (or a Slurm reso
 * Note you cannot use this option with the **exclude_list** option.
 * Note you cannot specify more than one node using this option.
 
-## What are valid root directories for the workflow?
-
-Currently the root directory should be be in your home directory and not on an NAS like `/nas/gaia/` as the submission will fail for an NFS reason.
-The experiment directory can be (and ought to be) on such a drive, though.
-
```
Comment (danielnapierski) [inline README.md:64]: Are there any restrictions on valid root dirs?  Should the README include a statement that root dirs can now be on shared /nas/gaia or /nas/user/xyz ?

Comment (LichMaster98) [reply README.md:64]: There are no longer any restrictions on valid root dirs. There used to be a NFS related bug when trying to mount a shared NAS as the root dir for Pegasus but it has been resolved.

Comment (LichMaster98) [reply README.md:64]: @danielnapierski I've added a note to the readme that the `root_dir` can be set to any location on the NAS, and that a shared drive e.g. `/nas/gaia` may be preferred due to space limitations on the user home dirs. However if unspecified the script will default to a users `/nas/home/xyz/` dir.

File: setup.py [hunk 2]
```diff
@@ -24,6 +24,8 @@
     install_requires=[
         "importlib-resources==1.4.0",
 	    "vistautils>=0.21.0",
+        "gitpython==3.1.12",
```
Comment (joecummings) [inline setup.py:27]: What is this being used for?

File: pegasus_wrapper/workflow.py [hunk 3]
```diff
@@ -243,14 +279,12 @@ def limit_jobs_for_category(self, category: str, max_jobs: int):
         """
         self._category_to_max_jobs[category] = max_jobs
 
-    def _conf_limits(self) -> str:
+    def _conf_limits(self) -> None:
         """
         Return a Pegasus config string which sets the max jobs per category appropriately.
```
Comment (joecummings) [inline pegasus_wrapper/workflow.py:284]: nit: not returning a string anymore so docstring should be updated

File: pegasus_wrapper/workflow.py [hunk 4]
```diff
@@ -171,55 +212,50 @@ def run_python_on_parameters(
...
-            self._added_files.add(checkpoint_pegasus_file)
-
         # If the checkpoint file already exists, we want to add it to the replica catalog
         # so that we don't run the job corresponding to the checkpoint file again
-        if checkpoint_path.exists():
-            with self._replica_catalog.open("a+") as handle:
-                handle.write(
-                    f"{ckpt_name} file://{checkpoint_path} site={self._default_site}\n"
-                )
+        checkpoint_pegasus_file = self.create_file(
```
Comment (joecummings) [inline pegasus_wrapper/workflow.py:254]: This is cool.

**Review by joecummings — APPROVED** [review]
lgtm

File: README.md [hunk 5]
```diff
@@ -72,7 +66,8 @@ Partitions each have a max walltime associated with them. See the saga cluster w
 
 If you change code while a pipeline is runnning, the jobs will pick up the changes. This could be helpful if you notice an error and fix it before that code runs, but can also lead to some unexpected behavior.
 
-## `No module named 'Pegasus'`
+## `No module named 'Pegasus'` (Version 4.9.3)
+*This is believed to have been fixed for Pegasus Version 5. If this arrises please leave an issue*
 
 This is a weird one that pops up usually when first getting set up with Pegasus. First, if you see this please contact one of the maintainers (currently @joecummings or @spigo900). To fix this, install the following packages with these commands in this exact order - they are dependent on each other.
```
Comment (joecummings) [inline README.md:72]: Can remove this fix. Comment you added is enough.

**Pull request merged**